src/HDF5.jl

```diff
@@ -558,8 +557,8 @@ function h5readattr(filename, name::AbstractString)
 end
 
 # Ensure that objects haven't been closed
-Base.isvalid(obj::Union{File,Properties,Datatype,Dataspace}) = obj.id != -1 && h5i_is_valid(obj.id)
-Base.isvalid(obj::Union{Group,Dataset,Attribute}) = obj.id != -1 && obj.file.id != -1 && h5i_is_valid(obj.id)
+Base.isvalid(obj::Union{File,Properties,Datatype,Dataspace}) = obj.id != -1 && h5i_is_valid(obj)
```

Comment:
`isvalid` and `close` have a lot of overlap and some of the close functions below are doing redundant checks.